### ISSUE TITLE
I am getting error saying ‘Failed to append/convert file parts, please check if the drive is full or write protected’, Why?

### ISSUE THREAD

**Issue by arungansi** [subhra74/xdm#272]
I am using version 7.2.11.....Also this does not happen in all websites. It downloads from sites which requires authentication but the content is public....for e.g Youtube.  I am trying to download the online class video from Udemy website for my registered courses. The download works sometimes and also it was working seamlessly till about 10 days ago. But now it is giving the above error after the download is 100% completed and the next second it flashes the above error message.

**Comment by psycheloveX** [comment]
无法组装文件，请检查文件系统是否已满或写保护。

**Comment by RenierMkXXVII** [comment]
I also encountered this problem and remembered how some programs can name files with long file names while windows can't process the name when copy/pasting the file. So what I did was to use "Save as" and rename the file with a short name (like just "aa", without qoutes). Then click "resume", that should continue to rebuild the file (unless if your disk is indeed full). I also noticed that XDM currently can't open download files from the list if the file name has asian characters in it or maybe non-english in general, so that might be connected to the issue.

TLDR; "Save as" --> short file name --> "Resume"

**Comment by subhra74** [comment]
@psycheloveX   Did the suggestion worked for you?

**Comment by Aikatsui** [comment]
@subhra74 I also noticed that problem on xdm latest version (Ubuntu 20.04.01)
Save as then Rename the file worked for me but shouldn't have such restriction on application then we cannot even recognize manually if there're duplicate downloads. Furthermore, i noticed that mess with non-english file names (eg: video files)

**Comment by mozesj** [comment]
I noticed the problem on xdm Version 7.2.11 running on Ubuntu 20.04 focal , when trying to download from one of the social media sites. Other sites download and save as expected.

_Save as then Rename_ did not work for me as @Aikatsui suggests.

Please see log snippet after running jre/bin/java -jar xdman.jar

` 081030fd-ac6b-4711-bc2e-9b528c9baca3 ] @ffmpeg_args: -y
[ 081030fd-ac6b-4711-bc2e-9b528c9baca3 ] FFmpeg exit code: 30
[ 081030fd-ac6b-4711-bc2e-9b528c9baca3 ] java.io.IOException: FFmpeg failed
	at xdman.downloaders.hls.HlsDownloader.assemble(HlsDownloader.java:647)
	at xdman.downloaders.hls.HlsDownloader.chunkComplete(HlsDownloader.java:126)
	at xdman.downloaders.SegmentImpl.transferComplete(SegmentImpl.java:93)
	at xdman.downloaders.AbstractChannel.copyStream2(AbstractChannel.java:176)
	at xdman.downloaders.AbstractChannel.run(AbstractChannel.java:75)
	at java.base/java.lang.Thread.run(Unknown Source)
[ 081030fd-ac6b-4711-bc2e-9b528c9baca3 ] List updated
[ 081030fd-ac6b-4711-bc2e-9b528c9baca3 ] removed
`

**Comment by subhra74** [comment]
It might be caused by unsupported encryption , however I am working on fix
